src/batch.js

```diff
@@ -0,0 +1,62 @@
+const call    = require('ramda/src/call')
+const curry   = require('ramda/src/curry')
+const juxt    = require('ramda/src/juxt')
+const max     = require('ramda/src/max')
+const zipWith = require('ramda/src/zipWith')
+
+// batch :: { k: v } -> ([a] -> Promise [b]) -> a -> Promise b
+const batch = (opts={}, f) => {
+  const {
+    limit = Infinity,
+    wait  = 32
+  } = opts
+
+  let args     = []
+  let last     = 0
+  let rejects  = []
+  let resolves = []
+  let timeout  = 0
+  let uniq     = new Map()
```

Comment:
Since you're never iterating over the map, can be a `WeakMap` if you felt so inclined.

Reply:
Good idea, but a `WeakMap` can only have objects as keys.  No primitive values allowed.

Reply:
Whoops. You're right.